src/NuGet.Clients/VsConsole/PowerShellHost/PowerShellHost.cs

```diff
@@ -576,21 +580,36 @@ public bool Execute(IConsole console, string command, params object[] inputs)
             if (ComplexCommand.AddLine(command, out fullCommand)
                 && !string.IsNullOrEmpty(fullCommand))
             {
+                // create a new token source with each command since CTS aren't usable once cancelled.
+                _tokenSource = new CancellationTokenSource();
+                _token = _tokenSource.Token;
                 return ExecuteHost(fullCommand, command, inputs);
             }
 
             return false; // constructing multi-line command
         }
 
-        protected static void OnExecuteCommandEnd()
+        protected void OnExecuteCommandEnd()
         {
             NuGetEventTrigger.Instance.TriggerEvent(NuGetEvent.PackageManagerConsoleCommandExecutionEnd);
+
+            // dispose token source related to this current command
+            _tokenSource?.Dispose();
```

Comment:
Shared state should be cleaned/reverted back. Other methods like `GetPathExpansionsAsyncCore` do that. More than that I see an extra step happens there:
```c#
SetPropertyValueOnHost(CancellationTokenKey, _token);
```
Consider copying it too.